rasterio/transform.py

```diff
@@ -126,14 +151,32 @@ def get_index(x, y, transform, op=math.floor, precision=6):
 
     Returns
     -------
-    row : int
-        row index
-    col : int
-        col index
+    rows : list of ints
+        list of row indicies
+    cols : list of ints
+        list of column indicies
     """
 
-    eps = 10.0**-precision * (1.0 - 2.0*op(0.1))
-    fcol, frow = ~transform * (x + eps, y - eps)
-    col = int(op(fcol))
-    row = int(op(frow))
-    return row, col
+    single_x = False
+    single_y = False
+    if not isinstance(xs, collections.Iterable):
+        xs = [xs]
+        single_x = True
+    if not isinstance(ys, collections.Iterable):
+        ys = [ys]
+        single_y = True
+
+    rows = []
+    cols = []
+    for x, y in zip(xs, ys):
+        eps = 10.0**-precision * (1.0 - 2.0*op(0.1))
```

Comment:
Move `eps` out of loop. Also we should vary the value of this in tests so we have a good handle on its impact.